schema-nft-marketplace.graphql

```diff
@@ -0,0 +1,109 @@
+  BOBA
+  BSC # aka BNB Chain
+  CELO
+  COSMOS
+  CRONOS
+  MAINNET # Ethereum Mainnet
+  FANTOM
+  FUSE
+  HARMONY
+  JUNO
+  MOONBEAM
+  MOONRIVER
+  NEAR_MAINNET
+  OPTIMISM
+  OSMOSIS
+  MATIC # aka Polygon
+  XDAI # aka Gnosis Chain
+}
+
+enum NftStandard {
+    ERC721
+    ERC721A
+    ERC1155
+}
+
+enum SaleStrategy {
+    " Strategy that executes an order at a fixed price that can be taken either by a bid or an ask. "
+    StandardSaleForFixedPrice
+    " Strategy that executes an order at a fixed price that can be matched by any tokenId for the collection. "
+    AnyItemFromCollectionForFixedPrice
+    " Strategy that executes an order at a fixed price that can be matched by any tokenId in a set of tokenIds. "
+    AnyItemInASetForFixedPrice
+    " Strategy to launch a Dutch Auction for a token where the price decreases linearly until a specified timestamp and end price defined by the seller. "
+    DutchAuction
+    " Strategy to set up an order that can only be executed by a specific address. "
+    PrivateSale
+}
+
+type Marketplace @entity {
+    id: ID!
+    name: String!
+    network: Network!
+    collectionCount: Int!
+    " Cumulative sale volume (in ETH) "
+    cumulativeSaleVolumeETH: BigDecimal!
+    " Cumulative sale volume (in USD)"
+    cumulativeSaleVolumeUSD: BigDecimal!
+    " Revenue that goes to the marketplace protocol, aka protocol fee "
+    marketplaceRevenue: BigDecimal!
+    " Revenue that goes to creator, aka royalty fee"
+    creatorRevenue: BigDecimal!
+    " Sum of marketplaceRevenue and creatorRevenue "
+    totalRevenue: BigDecimal!
+    totalTransactionCount: Int!
+}
+
+type Collection @entity {
+    " Contract address "
+    id: ID!
+    name: String!
+    symbol: String!
+    nftStandard: NftStandard!
+    " Royalty fee rate in percentage. E.g. 2.5% should be 2.5 "
+    royaltyFee: BigDecimal!
+    totalSupply: BigInt!
+    " Number of tokens for sale in the marketplace "
+    listedCount: BigInt!
+    " Floor price (in ETH) "
+    floorPriceETH: BigDecimal!
+    " Floor price (in USD) "
+    floorPriceUSD: BigDecimal!
+    cumulativeSaleCount: Int!
+    " Cumulative sale volume (in ETH) "
+    cumulativeSaleVolumeETH: BigDecimal!
+    " Cumulative sale volume (in USD)"
+    cumulativeSaleVolumeUSD: BigDecimal!
+    " Revenue that goes to the marketplace protocol, aka protocol fee "
+    marketplaceRevenue: BigDecimal!
+    " Revenue that goes to creator, aka royalty fee"
+    creatorRevenue: BigDecimal!
+    " Sum of marketplaceRevenue and creatorRevenue "
+    totalRevenue: BigDecimal!
+    totalTransactionCount: Int!
+}
+
+type Transaction @entity {
+    id: ID!
+    timestamp: BigInt!
+    blockNumber: BigInt!
+    collection: Collection!
+    tokenId: BigInt!
+    " Price (in ETH) "
+    priceETH: BigDecimal!
+    " Price (in USD) "
+    priceUSD: BigDecimal!
+    strategy: SaleStrategy!
+    " Maker account address "
+    maker: String!
+    " Taker account address "
+    taker: String!
```

Comment:
i wonder if it makes more sense to have a `buyer` and `seller` field vs `maker` and `taker` since maker and taker doesn't give much information as to who is on the giving/receiving end of the NFT transaction. is there any situation that i'm not thinking of where the maker/taker is useful?

e.g. alice is the seller and bob is the buyer
- alice lists an NFT for a fixed price, bob buys it
  - alice is the maker and bob is the taker 
- bob bids on an NFT from alice, alice accepts it
  - alice is the taker and bob is the maker